### PR TITLE
 Added local storage to persist data

### PR DESCRIPTION
**Pull request by Rahul139-stack**

### CODE REVIEW

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/script.js [hunk 1]
```diff
@@ -100,7 +101,13 @@ $(document).ready(function (e) {
 
     // Data can be updated via TMS Connection.
     if (componentTopic && componentHost) {
-      component.setAttribute("id", componentTopic);
+      // component.setAttribute("id", componentTopic);
```
Comment (Harikarthyk) [inline application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/script.js:104]: Why did you comment on this line?

If you are commenting on this, LIne 36 won't work, did you try this code?

Please check them again.

**Review by Harikarthyk — CHANGES_REQUESTED** [review]

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/script.js (hunk 1 above)
Comment (Harikarthyk) [inline application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/script.js:104]: Why is this line commented on?
Is there any reason?

If you comment on this then Line 36 will not work.

**Review by Harikarthyk — CHANGES_REQUESTED** [review]

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/script.js [hunk 2]
```diff
@@ -101,6 +102,13 @@ $(document).ready(function (e) {
     // Data can be updated via TMS Connection.
     if (componentTopic && componentHost) {
       component.setAttribute("id", componentTopic);
+      var localStorageValue = window.localStorage.getItem(`${componentTopic}`);
+      if (localStorageValue) {
+        updateTickerComponent(JSON.parse(localStorageValue).data, component);
+        window.reload()
```
Comment (Harikarthyk) [inline application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/script.js:108]: why are you reloading the entire page ?

Comment (Harikarthyk) [reply application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/script.js:108]: Is there any other solution?

**Review by Harikarthyk — CHANGES_REQUESTED** [review]

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/script.js (hunk 1 above)
Comment (Rahul139-stack) [reply application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/script.js]: Resolved

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/script.js (hunk 2 above)
Comment (Rahul139-stack) [reply application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/script.js]: Resolved

**Comment by wildone** [comment]
is this relevant?

**Comment by Harikarthyk** [comment]
Not Required any more

**Pull request closed without merging**